lib/Service/VerdictService.php

```diff
@@ -141,6 +141,7 @@ private function tagFile(int $fileId, string $tagName) {
 	 */
     public static function isFileTooLargeToScan(string $path): bool {
         $size = filesize($path);
+        if ($size === 0) { return false; }
         return !$size || $size > self::MAX_FILE_SIZE;
```

Comment:
```suggestion
        return $size > self::MAX_FILE_SIZE;
```

Reply:
Since the function can return a boolean false on failure this check should be kept

Reply:
Yeah. But !size does check for both. 0 and false.

Reply:
Just to be clear. If you want to have an explicit check for false `!$size` wont cut it. In that case I would suggest.

```php
if ($size === 0 || $size === false) { return false; }
return $size > self::MAX_FILE_SIZE;
```

Reply:
Yes it does. !size leads to our function returns true on failure. So if the filesize can not be evaluated we won't scan it because something is corrupt because the result of filesize() is only false on failure. In that case we want to return true for "wont scan".

Reply:
Ok, I see that. It would be a behavior change. But then I still would prefer `($size === false)` instead of `!$size` to make it more clear what's going on here.

Reply:
I have no preference for the syntax, feel free to make a suggestion.

Reply:
And it would still be a good case for a unit test. The [activity app uses phpunit](https://github.com/nextcloud/activity/tree/master/tests) as we also do in the vaas sdk.

Reply:
> I have no preference for the syntax, feel free to make a suggestion.

its not a question of syntax is a question of explicit type check.

```php
<?php

$foobar = null;
echo var_export(!$foobar); // true
echo var_export(!$foobar === null); // false

$foobar = 0;
echo var_export(!$foobar); // true
echo var_export(!$foobar === false); // false

$foobar = false;
echo var_export(!$foobar); // true
echo var_export($foobar === false); // true
```

Reply:
Doing this, you wouldn't even need you extra if statement because 0 is never > self::MAX_FILE_SIZE;

Reply:
That's why explicit checks are better most of the time.

Reply:
So far, we have classic PHPUnit tests outside of BATS. I would suggest to open a separate issue that implements some classic PHPUnit tests. I think your suggestion is quite good. However, I would quickly push this bugfix through for all versions because the bug was found by a stakeholder and not by us.

Reply:
So instead of 

```php
        if ($size === 0) { return false; }
        return !$size || $size > self::MAX_FILE_SIZE;
```

Just do this
```php
        return ($size === false) || $size > self::MAX_FILE_SIZE;
```

Reply:
Yes I am fine with that. Can you make this a code suggestion? I tested it and iT WoRkS LoCallY.

Reply:
You cant make code suggestions for multiple lines, so I would either do it on my local system or you do it.